app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -52,6 +52,119 @@ def build_config_spec(task_options)
       end
       set_spec_option(vmcs, :memoryMB, task_options[:vm_memory],      :to_i)
       set_spec_option(vmcs, :numCPUs,  task_options[:number_of_cpus], :to_i)
+
+      if task_options[:disk_remove] || task_options[:disk_add]
+        with_provider_object do |vim_obj|
+          task_options[:disk_remove].each { |d| remove_disk_config_spec(vim_obj, vmcs, d) } if task_options[:disk_remove]
+          add_disks(vim_obj, vmcs, task_options[:disk_add]) if task_options[:disk_add]
+        end
+      end
+    end
+  end
+
+  def add_disks(vim_obj, vmcs, disks)
+    ck, un = vim_obj.send(:getScsiCandU)
+
+    # if there is no scsi controller
+    if ck.nil?
+      ck, un = [-99, 0]
+      add_scsi_controller(vmcs, 0, ck)
+    end
+
+    disks.each do |d|
+      d[:controller_key] = ck
+      d[:unit_number]    = un
+      add_disk_config_spec(vmcs, d)
+      un += 1
+    end
+  end
+
+  def add_scsi_controller(vmcs, bus_number, dev_key)
+    device_type = 'VirtualLsiLogicController'
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Add) do |vdcs|
+      vdcs.device = VimHash.new(device_type) do |vDev|
+        vDev.sharedBus = VimString.new('noSharing', 'VirtualSCSISharing')
+        vDev.busNumber = bus_number
+        vDev.key       = dev_key
+      end
+    end
+  end
+
+  def backing_filename(diskname, unit_number)
+    filename = (diskname.blank? ? "disk_#{Time.zone.now.iso8601.gsub(/[-:TZ]/, '')}" : diskname) + "_#{unit_number}.vmdk"
+
+    primary_diskfile  = hardware.disks.find_by(:device_type => 'disk').try(:filename)
```

Comment:
You want to get the first disk (normally location 0:0) which means you need to `order` the results:
`hardware.disks.order(:location).find_by(:device_type => 'disk').try(:filename)`